lib/core.js

```diff
@@ -300,13 +300,21 @@ interface IteratorResult<T> {
     value?: T;
 }
 
-interface Iterator<T> {
-    next(): IteratorResult<T>;
-    @@iterator(): Iterator<T>;
+interface $Iterator<Y,R,N> {
+    @@iterator(): $Iterator<Y,R,N>;
+    next(value?: N): IteratorResult<Y|R>;
 }
+type Iterator<T> = $Iterator<T,void,void>;
 
-interface Iterable<T> {
-    @@iterator(): Iterator<T>;
+interface $Iterable<Y,R,N> {
```

Comment:
Do you have an opinion on the more explicit names? `Yield`, `Return`, and `Next`?